docs/Developers/Git_Basics.rst

```diff
@@ -539,6 +539,69 @@ commands (described below) for each commit to be rebased. These commands
 determine how individual commits will be transferred to the new base. You can
 also reorder the commit listing to change the order of the commits themselves.
 
+
+::
+
+    $ git reflog
+
+Will present a list of all actions performed on git and across all branches.
+On running ``git reflog`` each command performed till now will be presented
+on the terminal and each command will have an index. The structure of the
+index is ``HEAD@{index}``. This command is very important when you are in a
+mess. The next command will provide the users a weapon to fix their mistakes.
+
+
+::
+
+   $ git reset HEAD@{index}
+
+Reset to the point before you started messing up. ``HEAD@{index}`` denotes the
+index of the command before the command which broke everything.
+
+
+::
+
+   $ git branch new-branch-name
+   $ git reset HEAD~ --hard
+   $ git checkout new-branch-name
+
+These commands are useful if you have accidently commited something to master
+but should have been commited to a new branch. ``git branch new-branch-name``
+creates a new branch from the current state of the master.
+``git reset HEAD~ --hard`` removes the commit from the master branch.
+After running ``git checkout new-branch-name`` commit into this branch.
+Note:this does not work if you have already pushed to origin or run other
+commands before trying this.Then you have to use ``git reset HEAD@{index}``.
+
+
+::
+
+   $ git reset HEAD~ --soft
+   $ git stash
+   $ git checkout correct-branch
+   $ git stash pop
+   $ git add <file/folder_name>
+   $ git commit -m "your message"
+
+These series of commands are used when you have accidentally commited to the
+wrong branch.``git reset HEAD~ --soft`` undos the last commit and
```

Comment:
<img width="980" alt="screen shot 2018-01-14 at 23 18 28" src="https://user-images.githubusercontent.com/23073061/34921362-5c4cd4dc-f981-11e7-957e-b6559a9ebd27.png">
Just asking: Is it possible to do something with that?